<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="utf-8">
		<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
		<meta name="viewport" content="width=device-width, initial-scale=1">
		
		<meta name="author" content="Yanyu Liang">
		
		<meta name="generator" content="Hugo 0.31.1" />
		<title>Detecting epistasis in human complex traits &middot; Note Archive</title>
		<link rel="shortcut icon" href="/notebook/images/favicon.ico">
		<link rel="stylesheet" href="/notebook/css/style.css">
		<link rel="stylesheet" href="/notebook/css/highlight.css">

		
		<link rel="stylesheet" href="/notebook/css/font-awesome.min.css">
		

		

		
	</head>

    <body>
       <nav class="main-nav">
	
	
		<a href='/notebook/'> <span class="arrow">←</span>Home</a>
	
	<a href='/notebook/posts'>Archive</a>
	<a href='/notebook/tags'>Tags</a>
	<a href='/notebook/about'>About</a>

	

	
	<div class="searchbox">
    <label for="search-by"><i class="fa fa-search"></i></label>
    <input data-search-input id="search-by" type="text" placeholder="">
    <span data-search-clear=""><i class="fa fa-close"></i></span>
</div>
<script type="text/javascript" src="/notebook/js/lunr.min.js"></script>
<script type="text/javascript" src="/notebook/js/jquery-2.x.min.js"></script>
<script type="text/javascript" src="/notebook/js/horsey.js"></script>
<script type="text/javascript">
    var baseurl = "\/notebook\/";
</script>
<script type="text/javascript" src="/notebook/js/search.js"></script>

</nav>


        <section id="wrapper" class="post">
            <article>
                <header>
                    <h1>
                        Detecting epistasis in human complex traits
                    </h1>
                    <h2 class="headline">
                    Dec 30, 2017 10:30
                    · 1761 words
                    · 9 minutes read
                      <span class="tags">
                      
                      
                          
                              <a href="/notebook/tags/epistasis">epistasis</a>
                          
                              <a href="/notebook/tags/gwas">gwas</a>
                          
                              <a href="/notebook/tags/complex-trait">complex trait</a>
                          
                      
                      
                      </span>
                    </h2>
                </header>
                
                  
                

                <section id="post-body">
                  
                    <div id="TOC">
<ul>
<li><a href="#meta-data-of-reading">Meta data of reading</a></li>
<li><a href="#motivation">Motivation</a></li>
<li><a href="#methods-for-detecting-epistasis">Methods for detecting epistasis</a></li>
<li><a href="#overview-of-empirical-evidence-for-epistasis">Overview of empirical evidence for epistasis</a></li>
<li><a href="#summary">Summary</a></li>
</ul>
</div>

<p><span class="math display">\[
\newcommand\E{\text{E}}
\newcommand\nocr{\nonumber\\}
\newcommand\var{\text{Var}}
\newcommand\logit{\text{logit}}
\]</span></p>
<div id="meta-data-of-reading" class="section level1">
<h1>Meta data of reading</h1>
<ul>
<li><strong>Journal</strong>: Nature Review Genetics</li>
<li><strong>Year</strong>: 2014</li>
<li><strong>DOI</strong>: 10.1038/nrg3747</li>
</ul>
</div>
<div id="motivation" class="section level1">
<h1>Motivation</h1>
<p>Under GWAS framework, the causal variants are found with independent with additive and cumulative effects. But such assumption is arguably unrealistic. So it is necessary to consider the case where variants do not act independently (genetic interaction, epistasis). In this review, the authors refer:</p>
<ol style="list-style-type: decimal">
<li>Functional effect: <em>The general observation that effect of a particular variant depends on the genotype of another variant</em></li>
<li>Statistical effect: <em>The interaction variance that can be explained by a combination of causal variants that is not due to their independent effects</em> (namely, <span class="math inline">\(1 + 1 \ne 2\)</span>)</li>
</ol>
<p>, where <em>epistasis</em> is referred to statistical effect.</p>
<p>The authors point out that <em>the presence of functional epistasis does not automatically imply the presence of substantial statistical epistasis and vice versa</em>. By considering epistasis or not, there are two types of heritability:</p>
<ol style="list-style-type: decimal">
<li>Narrow-sense heritability (<span class="math inline">\(h^2\)</span>): The estimated proportion of the phenotypic variance of a trait that is <strong>attributable to independent, additive genetic effects</strong></li>
<li>Broad-sense heritability (<span class="math inline">\(H^2\)</span>): The estimated proportion of the phenotypic variance of a trait that is <strong>attributable to all additive and non-additive genetic effects</strong></li>
</ol>
<p>The authors point out the reason why epistasis is often under debate (its importance). There are two major goals of GWAS:</p>
<ol style="list-style-type: decimal">
<li>Identify causal variants of a particular trait to understand underlying biology</li>
<li>Estimate the effects of causal variants to predict phenotypic outcomes</li>
</ol>
<p>Functional epistasis can benefit the first goal and statistical epistasis can benefit the second goal.</p>
<p>This paper surveys the methodology and software of the detection of epistasis along with the latest empirical evidence for the importance of epistasis and potential utility in searching for genetic interaction.</p>
</div>
<div id="methods-for-detecting-epistasis" class="section level1">
<h1>Methods for detecting epistasis</h1>
<p>There have been methods developed to <em>detect whether two or more loci differs from that predicted by their individual effects</em>. Most methods use SNPs for pairwise or higher-order interactions in GWAS data. There are two strategies:</p>
<ol style="list-style-type: decimal">
<li>Hypothesis-free: Namely exhaustive search. Test for all possible combinations in all SNPs which involves billions of tests. It is both computationally expensive and statistically challenging</li>
<li>Hypothesis-driven: Test for a subset of SNPs and/or epistasis</li>
</ol>
<div id="regression-based-method" class="section level2">
<h2>Regression-based method</h2>
<p>The idea is to compare the saturated model (<span class="math inline">\(L_{S}\)</span>) and reduced model (<span class="math inline">\(L_R\)</span>) (considering interaction term between SNPs or not). For instance, in case-control study, <span class="math inline">\(L_S\)</span> can be:</p>
<span class="math display">\[\begin{align}
    \logit(Pr(Y = 1)) &amp;= \alpha  + \beta x_1 + \gamma x_2 + \delta x_1 x_2 \nonumber
  \end{align}\]</span>
<p>, where <span class="math inline">\(x_1, x_2\)</span> are binary variables and <span class="math inline">\(\delta\)</span> can be parameterized by four numbers according to the four possible configuration of <span class="math inline">\(x_1, x_2\)</span>.</p>
<p>Several progresses have been made to overcome the computational load. First, advanced data structure and parallelization have been introduced. Second, approximate tests have been applied, such as F ratio and Kirkwood superposition approximation (for <span class="math inline">\(L_S\)</span> versus <span class="math inline">\(L_R\)</span> tests under the assumption of HWE).</p>
<p>Although the hypothesis-free regression-based methods have become computationally tractable, they still suffer from low power (needs big sample size). Therefore, the practical compromise is to focus on the SNPs with genome-wide association signal. Moreover, the power of the test is a function of: i) interaction effect size; ii) sample size; iii) linkage disquilibrium. So, dense marker data can benefit the analysis as well.</p>
</div>
<div id="ld--and-haplotype-based-methods" class="section level2">
<h2>LD- and haplotype-based methods</h2>
<p>The idea is to test whether the co-occurrence of SNPs is enriched in cases relative to controls. It is computationally faster and statistically more powerful than regression-based methods. It works well for unlinked loci in rare diseases.</p>
<p>For instance, the LD of the pair of SNPs can be compared for cases and controls to see if the difference is significant (LD-based method). Haplotype-based methods adapted from LD-based one with equal power. With GWAS data, the linkage phase should be inferred before performing this test.</p>
<p>LD-based methods may generate inflated false positives because HWE does not always hold genome-wide. In LD-based method, a Z-score statistic is used from the difference in Pearson correlation in cases and controls. But simulation suggests that Z-score is inflated when two SNPs are highly correlated and/or both have significant marginal effects.</p>
<p>Haplotype-based methods incorporates a weighted average of the joint effects of two SNPs can control false positives when only one SNPs has marginal effects. This limitation can be overcome by using full logistic regression model (also with correction of covariates). So, a two-step approach where i) genome-wide screening with Z-score statistic (high power); ii) regression-based model to test the most promising interactions (low false positives).</p>
</div>
<div id="bayesian-methods" class="section level2">
<h2>Bayesian methods</h2>
<p>The idea is to compute the posterior probability for the SNP to be i) unassociated; ii) associated by marginal effect; iii) associated by joint effect. Combining Bayesian framework and GLM allows tests of SNP interactions with the consideration of covariates, marginal effect, and gene-environment interaction. Also, Bayesian approach can average multiple models when the underlying interaction patterns are unknown.</p>
</div>
<div id="data-filtering-methods" class="section level2">
<h2>Data-filtering methods</h2>
<p>Variance heterogeneity refers to the case where for a single bi-allelic SNP, the three genotypes have difference conditional phenotic variance. It can be used to select potentially interacting SNPs (since variance heterogeneity is a necessary condition for genetic interaction). But it suffers limited detectable variance heterogeneity.The cautions are:</p>
<ol style="list-style-type: decimal">
<li>Biases introduced by algorithm</li>
<li>Publication bias in existing knowledge</li>
<li>Context dependence</li>
<li>Filtering threshold may change the null distribution of test statistic</li>
</ol>
</div>
<div id="artificial-intelligence-algorithms" class="section level2">
<h2>Artificial intelligence algorithms</h2>
<p>…</p>
</div>
<div id="group--and-module-based-methods" class="section level2">
<h2>Group- and module-based methods</h2>
<p>The idea is to group SNPs into functional modules before performing tests, which can substantially reduce the statistical burden. A common practice is to group SNPs by gene and <em>derive gene-based variable that factors in SNP-based correlations for regression-based methods or for analogues LD-based approaches</em>. Another strategy is to compute pairwise interactions between groups and derive gene-based interaction p-value by integrating all pairwise p-values. Additionally, with pairwise interaction statistics between groups along with marginal effects, the gene-gene interaction can be analyzed using network analysis algorithms (SNPrank).</p>
<p>The authors point out that such gene-based method benefits from imputed genotypes from external LD information since it may capture unobserved causal variant. The implicit assumption behind gene-based method is “no intragenic interaction”.</p>
</div>
<div id="multitrait-and-multilevel-integration" class="section level2">
<h2>Multitrait and multilevel integration</h2>
<p>Pleiotropic epistasis is <em>statical interaction signals shared in multiple traits</em>. It can be identified by looking for SNP-SNP interaction that are shared across multiple traits or introducing composed “trait” (combining multiple traits) as latent variable in Bayesian framework. For example, a BEAM-derived method uses three latent variables: gene expression, SNPs, and individuals.</p>
</div>
<div id="summary-and-future-directions" class="section level2">
<h2>Summary and future directions</h2>

<figure >
    
        <img src="/notebook/images/epistasis.png" />
    
    
    <figcaption>
        <h4>Summary of reviewed methods for detecting epistasis</h4>
        
    </figcaption>
    
</figure>

<p>All methods mentioned above take genotyped SNPs and they cannot handle imputed SNPs with uncertainty. But this is useful in meta-analysis of epistasis which needs to be developed. Current methods do not consider sex chromosome.</p>
</div>
</div>
<div id="overview-of-empirical-evidence-for-epistasis" class="section level1">
<h1>Overview of empirical evidence for epistasis</h1>
<div id="hypothesis-free-studies" class="section level2">
<h2>Hypothesis-free studies</h2>
<p>People have done genome-wide epistasis scanning using WTCCC data. It turns out that the majority of the statistical interactions were in MHC region affecting T2D or RA. Such signals may be caused by haplotype effects where statistical interaction pair together tags a causal variant in proximity.</p>
<p>The result also indicated that many epistasis effect has multiplicative pattern, namely the effect of marginal additive term is bigger than the expected, and such effect can be removed by changing the scale of the trait measured (scale effect).</p>
<p>But scale effect happens in two marginally non-significant SNPs, which indicates that epistasis analysis may increase the power of marginal effect analysis. It is possible when two SNPs has small effect but their interaction term has big effect. Taking this idea, researchers used additive <span class="math inline">\(\times\)</span> additive model to scan the genome and identified similar results as the above one.</p>
<p>For traits as gene expression which have bigger genetic effect, researchers used BSGS data and identified 501 epistasis effect in discovery stage but only 30 could be replicated. The issue came from the unobserved causal variants (as discussed previously), which can drive the epistasis signal. The problem cannot be fixed even after filtering on LD. One important conclusion was that even corrected for power the attributable additive effects was still far more bigger than non-additive ones. The authors thought that since the current studies show few evidence of replicated non-additive effects in genome-wide analysis, the non-additive effect with large effect size is unlikely to exist.</p>
</div>
<div id="hypothesis-driven-studies" class="section level2">
<h2>Hypothesis-driven studies</h2>
<p>The hypothesis-driven studies can reduce the statistical burden substantially. The authors point out that the hypothesis-driven results also suffer from lack of replication.</p>
<p>The authors list several successful examples. In Alzheimer’s disease, <em>FYN</em> and <em>RNF219</em> decrease the risk only if <em>APOE4</em> mutant exists. Here the strategy is to limit the search to genetic effect that affects <em>endophenotypes</em>. Endophenotype is referred as the <em>heritable traits that are genetically correlated with disease traits. They are often traits (such as the level of a metabolite or transcript) that can be measured in all individuals (both diseased and healthy) and that can potentially provide a predictor of disease status</em>.</p>
<p>Two interacting SNPs in <em>HLA-DR2</em> affects the risk of multiple sclerosis toghether. The epistasis effect has been confirmed experimentally.</p>
<p>Regarding the pattern of interaction. If variant A increases risk only when variant B exists, the underlying mechanism might be that there are two redundant pathways (A in one and B in the other). Another pattern is that variant A increases risk only when variant B absents. It corresponds to the case where A and B are in the same pathway. Therefore, one strategy to propose the testing pairs is to make use of knowledge on biological function.</p>
<p>Alternatively, one can only test on variants with large marginal effect. There are cases where this strategy works. But it can also fail (no significant results for T2D, BMI, serum uric acid levels under current results).</p>
<p>The authors mention the pitfall of the current epistasis studies. They point out that the analysis is done on observed scale and an interaction that is non-additive in observed scale can be addtive in liability scale since even if variants contribute additively, if the disease occurs under some threshold, it appears to be epistasis under current analysis. It turns out that such effect is largest when disease prevalence is small.</p>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>Little evidence of statistical epistasis comparing to additive model and the genetic contribution substantially smaller than the additive one, especially for the case of pairwise epistasis between SNPs. However, hypothesis-driven analysis (based on biological function) is more successful than the genome-wide scanning one . It indicates that functional epistasis does exist. But the current progress has made genome-wide scanning (hypothesis-free tests) easily achievable.</p>
<p>The future direction is to develope meta-analysis method and to consider multilocus epistasis.</p>
</div>

                </section>
            </article>

            

            

            

            <footer id="footer">
    
    <p class="small">
    
       © Copyright 2017 <i class="fa fa-heart" aria-hidden="true"></i> Yanyu Liang
    
    </p>
    <p class="small">
        Powered by <a href="http://www.gohugo.io/">Hugo</a> Theme By <a href="https://github.com/nodejh/hugo-theme-cactus-plus">nodejh</a>
    </p>
</footer>

        </section>

        <script type="text/javascript" async
src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'],  ['\\[','\\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
  });
  MathJax.Hub.Queue(function() {
    
    
    
    var all = MathJax.Hub.getAllJax(), i;
    for(i = 0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
    }
  });

  MathJax.Hub.Config({
  
  TeX: { equationNumbers: { autoNumber: "AMS" } }
  });
</script>

<link href="/notebook/mermaid/mermaid.css" type="text/css" rel="stylesheet"/>
<script src="/notebook/mermaid/mermaid.js"></script>
<script>mermaid.initialize({startOnLoad:true});</script>







    </body>
</html>
