<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="utf-8">
		<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
		<meta name="viewport" content="width=device-width, initial-scale=1">
		
		<meta name="author" content="Yanyu Liang">
		
		<meta name="generator" content="Hugo 0.32.4" />
		<title>From Peas to Disease: Modifier Genes, Network Resilience, and the Genetics of Health &middot; Note Archive</title>
		<link rel="shortcut icon" href="/notebook/images/favicon.ico">
		<link rel="stylesheet" href="/notebook/css/style.css">
		<link rel="stylesheet" href="/notebook/css/highlight.css">

		
		<link rel="stylesheet" href="/notebook/css/font-awesome.min.css">
		

		

		
	</head>

    <body>
       <nav class="main-nav">
	
	
		<a href='/notebook/'> <span class="arrow">←</span>Home</a>
	
	<a href='/notebook/posts'>Archive</a>
	<a href='/notebook/tags'>Tags</a>
	<a href='/notebook/about'>About</a>

	

	
	<div class="searchbox">
    <label for="search-by"><i class="fa fa-search"></i></label>
    <input data-search-input id="search-by" type="text" placeholder="">
    <span data-search-clear=""><i class="fa fa-close"></i></span>
</div>
<script type="text/javascript" src="/notebook/js/lunr.min.js"></script>
<script type="text/javascript" src="/notebook/js/jquery-2.x.min.js"></script>
<script type="text/javascript" src="/notebook/js/horsey.js"></script>
<script type="text/javascript">
    var baseurl = "\/notebook\/";
</script>
<script type="text/javascript" src="/notebook/js/search.js"></script>

</nav>


        <section id="wrapper" class="post">
            <article>
                <header>
                    <h1>
                        From Peas to Disease: Modifier Genes, Network Resilience, and the Genetics of Health
                    </h1>
                    <h2 class="headline">
                    Jan 1, 2018 21:36
                    · 969 words
                    · 5 minutes read
                      <span class="tags">
                      
                      
                          
                              <a href="/notebook/tags/epistasis">epistasis</a>
                          
                              <a href="/notebook/tags/complex-trait">complex trait</a>
                          
                      
                      
                      </span>
                    </h2>
                </header>
                
                  
                

                <section id="post-body">
                  
                    <div id="TOC">
<ul>
<li><a href="#meta-data-of-reading">Meta data of reading</a></li>
<li><a href="#in-brief">In brief</a></li>
<li><a href="#modifier-genes">Modifier genes</a></li>
<li><a href="#modifier-effects-and-functional-significance">Modifier effects and functional significance</a></li>
<li><a href="#from-modified-phenotypes-to-modifier-genes">From modified phenotypes to modifier genes</a></li>
<li><a href="#features-of-modifier-genes">Features of modifier genes</a></li>
<li><a href="#modifiers-as-mediators-of-network-resilience">Modifiers as mediators of network resilience</a></li>
<li><a href="#summary">Summary</a></li>
</ul>
</div>

<div id="meta-data-of-reading" class="section level1">
<h1>Meta data of reading</h1>
<ul>
<li><strong>Journal</strong>: AJHG</li>
<li><strong>Year</strong>: 2017</li>
<li><strong>DOI</strong>: 10.1016/j.ajhg.2017.06.004</li>
</ul>
</div>
<div id="in-brief" class="section level1">
<h1>In brief</h1>
<p>This paper reviewed <em>the principles of modifier genetics and assess progress in the studies of modifier genes and their targets in both simple and complex traits</em>. They proposed that the modifier effects come from gene interaction network which depends on genetic background and it is of great use for *prevent, stabilize, and reverse disease and dysfunction.</p>
</div>
<div id="modifier-genes" class="section level1">
<h1>Modifier genes</h1>
<p>The causal gene does not act in isolation so the phenotypic outcome depends not only on identified causal variants but also on some related genetic variants (here “related” often means “related in functional network”). Those “related” genes are called modifier genes and they are often the results of: i) physical interaction; ii）mechanistic contribution (i.e. involved in the same biological pathway); iii) functional compensation (i.e. the modifier gene acts in redundant/alternative pathway of the causal one).</p>
<p>To experimentally detect modifier genes, we need the common functional variants with similar environmental factors in two populations with distinct genetic background.</p>
</div>
<div id="modifier-effects-and-functional-significance" class="section level1">
<h1>Modifier effects and functional significance</h1>
<div id="principles-of-modifier-genetics" class="section level2">
<h2>Principles of modifier Genetics</h2>

<figure >
    
        <img src="/notebook/images/modifier.png" />
    
    
    <figcaption>
        <h4>The illustration of modifier gene</h4>
        
    </figcaption>
    
</figure>

<p>The effect of modifier and target genes on phenotype are not additive (i.e. there is genetic interaction between the two). As shown in the above figure, modifier itself would not lead to phenotype. Taking another view, phenotype is the outcome of the integration of not only so called disease-causing genes but also environmental context and genetic background.</p>
</div>
<div id="types-of-target-genes" class="section level2">
<h2>Types of target genes</h2>
<p>It can be single target gene ot multigenic but the number of modifier-affected genes are largely unknown thanks to the lack of knowledge of genetic architecture and gene regulatory network.</p>
</div>
<div id="types-of-modifier-effects" class="section level2">
<h2>Types of modifier effects</h2>
<ol style="list-style-type: decimal">
<li><strong>Penetrance</strong>: The modifier gene changes the penetrance of the disease-causing variant</li>
<li><strong>Expressivity</strong>: The modifier gene changes the expressivity of the disease-causing variant</li>
<li><strong>Dominance</strong>: Dominance measures the genetic dosage required for a phenotype (complete dominant means heterozygous and homozygous have exactly the same phenotype). The modifier gene changes the dominance of the disease-causing variant</li>
<li><strong>Pleiotropy</strong>: When the modifier gene presents, it induces novel phenotypes</li>
</ol>
</div>
<div id="modifiers-of-complex-traits" class="section level2">
<h2>Modifiers of complex traits</h2>
<p>For single-gene phenotype, modifier study use multigenic background among a population sharing the same target variant. But for complex trait, only single modifier variant can be tested in a genetically clean population for differential phenotype. Note that the modifier effect is also context-dependent.</p>
</div>
</div>
<div id="from-modified-phenotypes-to-modifier-genes" class="section level1">
<h1>From modified phenotypes to modifier genes</h1>
<div id="to-identify-genetic-modifier-in-humans" class="section level2">
<h2>To identify genetic modifier in humans</h2>
<p>Two strategies are i) Look for shared variants between individuals with disease-causing variants without disease; iii) Perform target sequencing on disease gene to identify candidate with causal variant but phenotypically unaffected.</p>
<p>Through sequencing, researchers have found the reason of low penetrance of a disease-causing variant in a family where the occurrence of a common variant along with the disease-causing variant reaches 100% penetrance. Also, another research reported that the over-expression of one gene is protective to a LoF variant of disease gene.</p>
<p>GWAS type of study takes the strategy that collect a population which share the same disease-causing variant but with various phenotypic outcome (severity, age of onset, etc). Also, the same strategy can be used in family-based study (linkage study) where the signal-noise ratio is better.</p>
</div>
<div id="to-identify-genetic-modifiers-in-mice" class="section level2">
<h2>To identify genetic modifiers in mice</h2>
<p>…</p>
</div>
</div>
<div id="features-of-modifier-genes" class="section level1">
<h1>Features of modifier genes</h1>
<div id="nature-of-sequence-variation" class="section level2">
<h2>Nature of sequence variation</h2>
<p>The number of identified modifier variants is small but it has suggested considerable heterogeneity. The following lists several outcomes of modifier variants:</p>
<ol style="list-style-type: decimal">
<li>Partial or complete loss of function (missense mutation or gene deletion)</li>
<li>Enhanced function (increased protein stability)</li>
<li>Gain of alternative function (novel protein-protein interaction)</li>
</ol>
<p>Currently discovered modifier variants often occur in protein-coding exon but may act as regulatory variant (UTRs or promoters). <strong>In general, any variant that affect gene function by affecting interaction with other genes and networks is modifier variant.</strong></p>
</div>
<div id="nature-of-modifier-functions" class="section level2">
<h2>Nature of modifier functions</h2>
<p><strong>One interesting question is that whether modifier genes tend to alter a general regulatory process or affect the target gene locally.</strong> It turns out that the latter is the case according to current data.</p>
</div>
<div id="mutational-buffering" class="section level2">
<h2>Mutational buffering</h2>
<p>Here the authors discussed a general mechanism to suppress the effect of deleterious mutations by chaperones. But there is not mapping study showing that chaperone is modifier gene, but it is clear that the change of chaperone can modify the effect of a causal variant.</p>
<p><strong>Global modifier</strong> refers to the genes that regulate the outcome of variant via general mechanisms, such as chaperones, chromatin modifiers, and transcriptional regulators.</p>
<p>It seems that global modifiers respond to generalized stress and modifier genes target specific dysfunctions.</p>
</div>
</div>
<div id="modifiers-as-mediators-of-network-resilience" class="section level1">
<h1>Modifiers as mediators of network resilience</h1>
<p>Modifier effect can be seen as an consequence of causal variant to compensate or enlarge the effect of the target variant on phenotype. Another view is that modifier effect indicates the network interaction (it can be molecularly direct interaction or related functionally, i.e. in the related pathway).</p>
<p>So, such coordination between target variant and modifier should be context-dependent (i.e. genetic background, environmental factors). The number of deleterious mutations per generation is hundreds in human including disease-causing ones but human is still alive. One possibility is that some genes act as the buffering system (latent modifiers) to counteract these deleterious variants in a genetic background-specific and environment-specific manner.</p>
<p>The complexity is that whether the gene is target, modifier, or QTL depends on the nature of the variant, genetic background, and environmental context. Network feature can provide substantial information but it is unavailable right now.</p>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>Modifier effect is pervasive and it works coordinately with target genes. Its occurrence indicates that the functionally related role between the target and modifier in gene network. Modifier effect provides a mechanism in the network to overcome the deleterious variant occurring naturally in individuals.</p>
</div>

                </section>
            </article>

            

            

            

            <footer id="footer">
    
    <p class="small">
    
       © Copyright 2018 <i class="fa fa-heart" aria-hidden="true"></i> Yanyu Liang
    
    </p>
    <p class="small">
        Powered by <a href="http://www.gohugo.io/">Hugo</a> Theme By <a href="https://github.com/nodejh/hugo-theme-cactus-plus">nodejh</a>
    </p>
</footer>

        </section>

        <script type="text/javascript" async
src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'],  ['\\[','\\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
  });
  MathJax.Hub.Queue(function() {
    
    
    
    var all = MathJax.Hub.getAllJax(), i;
    for(i = 0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
    }
  });

  MathJax.Hub.Config({
  
  TeX: { equationNumbers: { autoNumber: "AMS" } }
  });
</script>

<link href="/notebook/mermaid/mermaid.css" type="text/css" rel="stylesheet"/>
<script src="/notebook/mermaid/mermaid.js"></script>
<script>mermaid.initialize({startOnLoad:true});</script>







    </body>
</html>
