---
title: "Epistasis — the essential role of gene interactions in the structure and evolution of genetic systems"
date: 2018-01-03T09:30:59-06:00
author: "Yanyu Liang"
tags: ["epistasis in evolution"]
categories: ["biology - research review"]
draft: false
output:
  blogdown::html_page:
    toc: true
    toc_depth: 1
bibliography: patrick-nrg-2008.bib
---


<div id="TOC">
<ul>
<li><a href="#meta-data-of-reading">Meta data of reading</a></li>
<li><a href="#three-meanings-of-epistasis">Three meanings of epistasis</a></li>
<li><a href="#comments">Comments</a></li>
<li><a href="#epistasis-as-a-tool">Epistasis as a tool</a></li>
<li><a href="#epistasis-as-an-obstacle">Epistasis as an obstacle</a></li>
<li><a href="#structure-and-evolution-of-complex-systems">Structure and evolution of complex systems</a></li>
<li><a href="#conclusion">Conclusion</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<div id="meta-data-of-reading" class="section level1">
<h1>Meta data of reading</h1>
<ul>
<li><strong>Journal</strong>: Nature Review Genetics</li>
<li><strong>Year</strong>: 2008</li>
<li><strong>DOI</strong>: 10.1038/nrg2452</li>
</ul>
</div>
<div id="three-meanings-of-epistasis" class="section level1">
<h1>Three meanings of epistasis</h1>
<div id="functional-epistasis" class="section level2">
<h2>Functional epistasis</h2>
<p>Two genes are physically interacting with each other or functionally related to each other (i.e. in the same/redundant pathways).</p>
</div>
<div id="compositional-epistasis" class="section level2">
<h2>Compositional epistasis</h2>
<p>It refers to the deviation from Mendelian segregation ratio. According to the pattern of the skewed ratio, the order of the two genes can be identified. Note that the order is allele-specific, namely even the gene pair kept the same, different alleles may give different orders.</p>
</div>
<div id="statistical-epistasis" class="section level2">
<h2>Statistical epistasis</h2>
<p>It means that the observed effect of two alleles in two genes is significantly different from the expected one (on the basis of some pre-defined models). There are two widely used models, <span class="math inline">\(W_{xy} = \alpha_x + \alpha_y + \epsilon\)</span> (additive, simplified from dominance/recessive model, quantitative traits) and <span class="math inline">\(W_{xy} = \alpha_x \alpha_y + \epsilon\)</span> (multiplicative, fitness).</p>
</div>
</div>
<div id="comments" class="section level1">
<h1>Comments</h1>
<p>The paper mentioned the issue of multiplicative model, that is LD is stable <span class="citation">(Karlin and Feldman 1978)</span>. Therefore, a pair of genes in LD does not imply that they interact epistatically.</p>
<p>The difference between compositional epistasis and statistical epistasis is that the former uses a fixed genetic background but the latter uses a population where the genetic background is not identical.</p>
</div>
<div id="epistasis-as-a-tool" class="section level1">
<h1>Epistasis as a tool</h1>
<p>It cam be used to construct gene network. High-throughput techniques has been used to survey the epistatic pairs which influence the viability of yeast (mutation was limited to deletion). Then a genetic network can be built upon such information.</p>
<p>Besides deletion, over-expression has also been surveyed. But it needs to be noticed again that the epistatic effect is allele-specific.</p>
</div>
<div id="epistasis-as-an-obstacle" class="section level1">
<h1>Epistasis as an obstacle</h1>
<p>The author pointed out that the epistatic effect can cause the failure of GWAS study where only single-locus effect is considered. <span class="citation">(Carlborg et al. 2006)</span> provides an interesting story about how epistasis shape the phenotype under selection.</p>
<p>In the study, chicken were selected for different weights for 40 generations and the eQTL study was done but only one QTL was found (Growth9). However, five more additional signals were identified after considering epistasis (namly considering interaction term for each locus pair <span class="citation">(Carlborg and Haley 2004)</span>) as shown in the following figure.</p>
{{< figure src="/notebook/images/chicken.png" title="The results of the chicken study" >}}
<blockquote>
<p>Comments: Interestingly, for those of QTLs that act in epistatic way with others, the conditional phenotypic variation is different (see figure c above), which is another sign of epistasis signal.</p>
</blockquote>
<p>The author also mentioned that the cross-interpretation between GWAS signal and QTL-mapping is affected by epistasis if the populations are not matched for these two studies.</p>
<div id="epistasis-in-natural-population" class="section level2">
<h2>Epistasis in natural population</h2>
<p>The contribution of epistasis to complex traits (beyond the additive model) is remained unknown.</p>
</div>
<div id="epistasis-and-human-diseases" class="section level2">
<h2>Epistasis and human diseases</h2>
<p>Epistasis is one explanation of why disease-causing variants still persist in populations. The paper showed an example where two loci were found to be in natural selection-induced LD in multiple sclerosis.</p>
{{< figure src="/notebook/images/hla_mouse_model.png" title="Testing the epistatic effect in mice" >}}
<p>The epistatic effect was tested in mice (above figure) where one of the allele can partially rescue the effect of the other, which indicates epistasis. But in human, epistasis is hard to be tested since the underlying recombination rate between these two loci is unknown.</p>
</div>
<div id="inferring-epistatic-effect-in-gwas" class="section level2">
<h2>Inferring epistatic effect in GWAS</h2>
<p>To limit the analysis to loci with large effect size is not valid since epistatic interaction with large effect may have relatively some marginal effect (<em>epistasis is frequently detected in the absence of main effects</em>).</p>
</div>
</div>
<div id="structure-and-evolution-of-complex-systems" class="section level1">
<h1>Structure and evolution of complex systems</h1>
<p>The author pointed out that epistatic effect is a natural consequence of the nature selection of complex system since every evolutionary change is built upon the previous ones. The author mentioned three views of the generation of epistasis under natural selection (figure below).</p>
{{< figure src="/notebook/images/epistasis_selection.png" title="Three views of the generation of epistasis under natural selection" >}}
<p>Fisher’s model (figure a) says that epistasis is a by-product of evolutionary process where adaption is just a hill-climbing process and the average/additive effects are the driving force. Wright’s model (figure b) says epistasis affects the direction of evolution since epistasis determines the overall fitness landscape with multiple peaks and the genetic change is due to genetic drift. Gavrelet’s model (figure c) says that since the genotype space is high-dimensional, the fitness nearly equal locally (at least for some directions) without barriers in Wright’s model, therefore the change can either by directional selection or drift.</p>
<p>Nevertheless, <em>epistasis and genetic interaction are inevitable consequence of the evolutionary process</em>.</p>
<div id="epistasis-and-the-path-of-evolutionary-change" class="section level2">
<h2>Epistasis and the path of evolutionary change</h2>
<p>One important question in evolutionary biology is whether epistasis determines the path of evolutionary change. The interactions between distant loci were traditional focus. The author pointed out that result obtained in the research of epistasis effect among loci in a single protein <span class="citation">(Ortlund et al. 2007)</span>.</p>
</div>
<div id="the-evolution-of-regulatory-complexity" class="section level2">
<h2>The evolution of regulatory complexity</h2>
<p>It turns out that <em>linkage can facilitate the maintenance of epistatic interactions and vice versa</em>. It may also appear in the gene regulation in local genomic region (ENCODE paper was cited here but the result is not convincing to me, namely 5% of genomic region is under evolutionary constraint).</p>
</div>
</div>
<div id="conclusion" class="section level1">
<h1>Conclusion</h1>
<p>From evolutionary standpoint, one question is that whether the structure of the system has evolved to facilitate these network properties (modularity <span class="citation">(Segre et al. 2005)</span>).</p>
<blockquote>
<p>Here the modularity means that the genes in the same module tends to have the same sign of epistatic effect on the elements outside the module. Another point of view of modularity is that the functional pathways act independently with each other, in this case, the trait is more evolvable (less pleiotropy) <span class="citation">(Wagner, Pavlicev, and Cheverud 2007)</span>. Note that the former modularity is for single cell organism where the epistatic effect is limited within the cell. In contrast, multicellular organism may have inter-cellular epistatic effect as well.</p>
</blockquote>
<p>It is necessary to unify the three meanings of epistasis as mentioned at the beginning. <span class="citation">(Jansen 2003)</span> gave a try to make the connection between functional epistasis and statistical epistasis.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-carlborg2004epistasis">
<p>Carlborg, Örjan, and Chris S Haley. 2004. “Epistasis: Too Often Neglected in Complex Trait Studies?” <em>Nature Reviews Genetics</em> 5 (8). Nature Publishing Group: 618–25.</p>
</div>
<div id="ref-carlborg2006epistasis">
<p>Carlborg, Örjan, Lina Jacobsson, Per Åhgren, Paul Siegel, and Leif Andersson. 2006. “Epistasis and the Release of Genetic Variation During Long-Term Selection.” <em>Nature Genetics</em> 38 (4). Nature Publishing Group: 418–20.</p>
</div>
<div id="ref-jansen2003studying">
<p>Jansen, Ritsert C. 2003. “Studying Complex Biological Systems Using Multifactorial Perturbation.” <em>Nature Reviews Genetics</em> 4 (2). Nature Publishing Group: 145–51.</p>
</div>
<div id="ref-Karlin1978">
<p>Karlin, Samuel, and Marcus W Feldman. 1978. “SIMULTANEOUS STABILITY OF D=0 AND D≠0 FOR MULTIPLICATIVE VIABILITIES AT TWO LOCI.” <em>Genetics</em> 90 (4): 813 LP–825. <a href="http://www.genetics.org/content/90/4/813.abstract" class="uri">http://www.genetics.org/content/90/4/813.abstract</a>.</p>
</div>
<div id="ref-ortlund2007crystal">
<p>Ortlund, Eric A, Jamie T Bridgham, Matthew R Redinbo, and Joseph W Thornton. 2007. “Crystal Structure of an Ancient Protein: Evolution by Conformational Epistasis.” <em>Science</em> 317 (5844). American Association for the Advancement of Science: 1544–8.</p>
</div>
<div id="ref-segre2005modular">
<p>Segre, Daniel, Alexander DeLuna, George M Church, and Roy Kishony. 2005. “Modular Epistasis in Yeast Metabolism.” <em>Nature Genetics</em> 37 (1). Nature Publishing Group: 77–83.</p>
</div>
<div id="ref-wagner2007road">
<p>Wagner, Günter P, Mihaela Pavlicev, and James M Cheverud. 2007. “The Road to Modularity.” <em>Nature Reviews Genetics</em> 8 (12). Nature Publishing Group: 921–31.</p>
</div>
</div>
</div>
